bionic/deriver.py

```diff
@@ -542,23 +553,21 @@ def __init__(self, task):
         self.case_key = None
         self.provider = None
 
-        # These are set by
-        # EntityDeriver._set_up_provenance_for_task_state(), just
-        # before the task state becomes eligible for computation.
+        # These are set by EntityDeriver._complete_task_state(), just
+        # before the task state becomes eligible for cache lookup / computation.
```

Comment:
Maybe we could add `They will be present if and only if is_complete is True.`

Reply:
Yeah this makes sense. I said a similar thing about `hashed_results_by_name` in my reply to previous comment!